spring-kafka/src/main/java/org/springframework/kafka/core/StreamsBuilderFactoryBean.java

```diff
@@ -166,6 +179,7 @@ public void stop(Runnable callback) {
 	public synchronized void start() {
 		if (!this.running) {
 			try {
+				Assert.notNull(this.streamsConfig, "'streamsConfig' must not be null");
```

Comment:
Would be better to have this in the `setStreamsConfig()` and in the `createInstance()`.

The last one will work from the application  start up to notify end-user that he/she hasn't called  `setStreamsConfig()` at all.

And the first one won't let to set `null` at all.

It's not good to check it from the `start()` because it may happen on a different thread and the failure might mislead a bit from the initialization phase or re-initialization at runtime.